sled-agent/src/instance_manager.rs

```diff
@@ -97,44 +106,68 @@ impl InstanceManager {
         })
     }
 
-    /// Sets the VMM reservoir size to the requested (nonzero) percentage of
-    /// usable physical RAM, rounded down to nearest aligned size required by
-    /// the control plane.
+    /// Sets the VMM reservoir to the requested percentage of usable physical
+    /// RAM or to a size in MiB. Either mode will round down to the nearest
+    /// aligned size required by the control plane.
     pub fn set_reservoir_size(
         &self,
         hardware: &sled_hardware::HardwareManager,
-        target_percent: u8,
+        mode: ReservoirMode,
     ) -> Result<(), Error> {
-        assert!(
-            target_percent > 0 && target_percent < 100,
-            "target_percent {} must be nonzero and < 100",
-            target_percent
-        );
+        let hardware_physical_ram_bytes = hardware.usable_physical_ram_bytes();
+        let req_bytes = match mode {
+            ReservoirMode::None => return Ok(()),
+            ReservoirMode::Size(mb) => {
+                let bytes = ByteCount::from_mebibytes_u32(mb).to_bytes();
+                if bytes > hardware_physical_ram_bytes {
+                    return Err(Error::ReservoirConfig(format!(
+                        "cannot specify a reservoir of {bytes} bytes when \
+                        physical memory is {hardware_physical_ram_bytes} bytes",
+                    )));
+                }
+                bytes
+            }
+            ReservoirMode::Percentage(percent) => {
+                if !matches!(percent, 1..=99) {
+                    return Err(Error::ReservoirConfig(format!(
+                        "reservoir percentage of {} must be between 0 and 100",
```

Comment:
nit: "VMM reservoir" to be extra clear

Reply:
Fixed in [23a25d6](https://github.com/oxidecomputer/omicron/pull/4137/commits/23a25d6071e32d2cf4b14cb17cd744d3f30aa814)